src/compiler/transformer.ts

```diff
@@ -46,6 +46,14 @@ namespace ts {
     }
 
     /**
+     * Tracks a monotonically increasing transformation id used to associate a node with a specific
+     * transformation. This ensures transient properties related to transformations can be safely
+     * stored on source tree nodes that may be reused across multiple transformations (such as
+     * with compile-on-save).
+     */
+    let nextTransformId = 1;

```

Comment:
Can you comment on the lifetime of this transformId? You mentioned monotonically increasing but is that in a global context, or per compilation?